onnx2tf/utils/common_functions.py

```diff
@@ -685,44 +717,68 @@ def explicit_broadcast(
-        # # Check output values by brute force only if the shape of const_or_var_2
-        # # is different between const_or_var_1 and const_or_var_2 in dimensions other than 1
-        # const_or_var_1_shape_not_none = [
-        #     i if not isinstance(i, str) else 1 for i in const_or_var_1_shape
-        # ]
-        # for var1_shape, var2_shape in zip(const_or_var_1_shape_not_none, const_or_var_2.shape):
-        #     if var2_shape != 1 and var1_shape != 1 and var1_shape != var2_shape:
-        #         dummy_data_onnx = np.random.random_sample(graph_node.inputs[0].shape)
-        #         dummy_data_onnx = dummy_data_onnx.astype(graph_node.inputs[0].dtype)
-        #         dummy_data_tf = dummy_data_onnx.copy()
-        #         dummy_data_tf = dummy_data_tf.reshape(const_or_var_1_shape).astype(graph_node.inputs[0].dtype)
-        #         simple_arithmetic_validity_check(
-        #             op_type=graph_node.op,
-        #             onnx_x=dummy_data_onnx,
-        #             onnx_y=graph_node.inputs[1].values,
-        #             tf_x=dummy_data_tf,
-        #             tf_y=const_or_var_2,
-        #         )
-        #         break
-
-    elif isinstance(const_or_var_2, tf.Tensor) \
-        or (
-            not isinstance(const_or_var_2, np.ndarray) \
-            and tf.keras.backend.is_keras_tensor(const_or_var_2)
-        ):
-        if graph_node_input_name2 is not None \
-            and tf_layers_dict is not None \
-            and graph_node_input_name2 in tf_layers_dict \
-            and tf_layers_dict[graph_node_input_name2]['optype'] == 'Input':
-            const_or_var_2: np.ndarray = tf.transpose(
-                a=const_or_var_2,
-                perm=transpose_perm
-            )
-    else:
+        graph_node_input_shape2 = [1] + graph_node_input_shape2
+
+    # Swap operands to apply transpose to correct target if needed
+    # second operand is always target of transpose
+    if broadcast_validity_check(list(const_or_var_1.shape), graph_node_input_shape1) and \
+            not broadcast_validity_check(list(const_or_var_2.shape), graph_node_input_shape2):
+        const_or_var_1, const_or_var_2 = const_or_var_2, const_or_var_1
+        graph_node_input_name1, graph_node_input_name2 = graph_node_input_name2, graph_node_input_name1
+        graph_node_input_shape1, graph_node_input_shape2 = graph_node_input_shape2, graph_node_input_shape1
+        swapped += 1
+
+    # Check if operands need transpose
+    # CAUTION: this part may occur problem when there are more than two same numbers in tensor shape.
+    #          please consider manual debugging if output is differ with onnx.
+    if broadcast_validity_check(list(const_or_var_1.shape), list(const_or_var_2.shape)) and \
+            broadcast_validity_check(graph_node_input_shape1, graph_node_input_shape2):
         pass
+    else:
+        transpose_perm = [0] + [i+2 for i in range(len(const_or_var_1.shape)-2)] + [1]        
```

Comment:
Lint warning displayed when importing patches. If possible, please remove the unnecessary trailing whitespace.
```
transpose_perm = [0] + [i+2 for i in range(len(const_or_var_1.shape)-2)] + [1]        <----
```